### PR TITLE
UMLet 14.2 (new formula)

### PR DESCRIPTION
**Pull request by barthel**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

UMLet is a UML tool aimed at providing a fast way of creating UML diagrams.

### CODE REVIEW

File: Formula/umlet.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+class Umlet < Formula
+  desc "This UML tool aimed at providing a fast way of creating UML diagrams."
+  homepage "http://www.umlet.com/"
+  url "http://www.umlet.com/umlet_14_2/umlet-standalone-14.2.zip"
+  sha256 "dc09538d04cb899218f3bdfdb5205f60359cff68cb1428d1228c6d91743d8cb9"
+
+  bottle :unneeded
+
+  keg_only ""
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Please fill this out if there's a good reason for it.

File: Formula/umlet.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    # Replace '${programDir}'
+    inreplace "#{prefix}/umlet.sh", "${programDir}/umlet.jar", "#{prefix}/umlet.jar"
+
+    # Make original script file 'umlet.sh' executable
+    chmod 0755, "#{prefix}/umlet.sh"
+
+    # symlink shell script to 'bin/umlet-<version>'
+    (bin/"umlet-#{version}").write_env_script "#{prefix}/umlet.sh", Language::Java.java_home_env("1.6+")
+    # symlink shell script to 'bin/umlet'
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Don't need any comments in this function; they are all self-explanatory.

File: Formula/umlet.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    # java_classpath = Dir[prefix/"**/*.jar"].join(":")
+
+    # Replace 'java' command in shell script
+    inreplace "#{prefix}/umlet.sh", " java ", " ${JAVA_HOME}/bin/java "
+
+    # Deactivate use of 'programDir'
+    inreplace "#{prefix}/umlet.sh", "programDir=", "#programDir="
+
+    # Replace '${programDir}'
+    inreplace "#{prefix}/umlet.sh", "${programDir}/umlet.jar", "#{prefix}/umlet.jar"
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Can you please submit these patches to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patches are needed in a comment in the formula file. Thanks!

File: Formula/umlet.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    # symlink shell script to 'bin/umlet-<version>'
+    (bin/"umlet-#{version}").write_env_script "#{prefix}/umlet.sh", Language::Java.java_home_env("1.6+")
+    # symlink shell script to 'bin/umlet'
+    bin.install_symlink "umlet-#{version}" => "umlet"
+    bin.chmod 0755
+  end
+
+  test do
+    system "#{bin}/umlet", "-action=convert", "-format=png", "-output=#{testpath}/test-output.png", "-filename=#{prefix}/palettes/Plots.uxf"
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Wrap this at 80 characters.

File: Formula/umlet.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    inreplace "#{prefix}/umlet.sh", "${programDir}/umlet.jar", "#{prefix}/umlet.jar"
+
+    # Make original script file 'umlet.sh' executable
+    chmod 0755, "#{prefix}/umlet.sh"
+
+    # symlink shell script to 'bin/umlet-<version>'
+    (bin/"umlet-#{version}").write_env_script "#{prefix}/umlet.sh", Language::Java.java_home_env("1.6+")
+    # symlink shell script to 'bin/umlet'
+    bin.install_symlink "umlet-#{version}" => "umlet"
+    bin.chmod 0755
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Shouldn't be necessary; will be done after installation.

**Comment by mikemcquaid** [comment]
> Moving JAR files into libexec doesn't work while the classpath is hard-coded in MANIFEST file in umlet.jar.

We'll need to figure out how to avoid hard-coding this. We can't install jars into `lib`.

**Comment by barthel** [comment]
@mikemcquaid  Move installation into '''libexec''' and rework on all comments.

File: Formula/umlet.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  bottle :unneeded
+
+  depends_on :java => "1.6+"
+
+  def install
+    rm Dir["*.{desktop,exe}"]
+
+    libexec.install Dir["*"]
+
+    # Replace 'java' command in shell script for use the 'JAVA_HOME' env.
```
Comment (mikemcquaid) [inline Formula/umlet.rb]: Can remove this comment; self-explanatory from the code. Then we're good to pull I think 👍

Comment (barthel) [inline Formula/umlet.rb]: Done.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**